### ISSUE TITLE
panic: deletion not supported

### ISSUE THREAD

**Issue by vrogojin** [binance-chain/bsc#475]
Started fresh full node with snap sync
`geth --syncmode=snap --config ./config.toml --datadir node  --cache 18000 --rpc --http.port 9545 --http.corsdomain "*" --http.api "eth,web3,personal,net,txpool" --ws --ws.port 8546 --ws.origins "*" \
    --ws.api "eth,web3,personal,net,txpool" --rpc.allow-unprotected-txs --txlookuplimit 0 console`

After around an hour it crashes
```

INFO [10-23|00:29:36.318] Starting Geth on Ethereum mainnet...
INFO [10-23|00:29:36.321] Maximum peer count                       ETH=30 LES=0 total=30
INFO [10-23|00:29:36.321] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
WARN [10-23|00:29:36.321] Option nousb is deprecated and USB is deactivated by default. Use --usb to enable
Welcome to the Geth JavaScript console!

instance: Geth/v1.1.2-c4f93121-20210825/linux-amd64/go1.15.14
at block: 0 (Mon Apr 20 2020 21:46:54 GMT+0800 (+08))
 datadir: <my_data_dir_here>
 modules: admin:1.0 debug:1.0 eth:1.0 miner:1.0 net:1.0 parlia:1.0 personal:1.0 rpc:1.0 txpool:1.0 web3:1.0

To exit, press ctrl-d
> eth.syncing.
eth.syncing.constructor eth.syncing.toString eth.syncing.valueOf
> eth.syncing
{ 
  currentBlock: 0,
  highestBlock: 11997535,
  knownStates: 0,
  pulledStates: 0,
  startingBlock: 0
}
> net.peerCount
2
> net.peerCount
4
> panic: deletion not supported

goroutine 459309 [running]:
github.com/ethereum/go-ethereum/trie.(*StackTrie).TryUpdate(0xc069403520, 0xc077307a00, 0x20, 0x20, 0x2857fb0, 0x0, 0x0, 0xc019bb5698, 0xc07dd149a0)
        github.com/ethereum/go-ethereum/trie/stacktrie.go:194 +0xd2
github.com/ethereum/go-ethereum/trie.(*StackTrie).Update(0xc069403520, 0xc077307a00, 0x20, 0x20, 0x2857fb0, 0x0, 0x0)
        github.com/ethereum/go-ethereum/trie/stacktrie.go:201 +0x77
github.com/ethereum/go-ethereum/eth/protocols/snap.(*Syncer).processStorageResponse(0xc00000c780, 0xc061a85500)
        github.com/ethereum/go-ethereum/eth/protocols/snap/sync.go:1928 +0x3f4
github.com/ethereum/go-ethereum/eth/protocols/snap.(*Syncer).Sync(0xc00000c780, 0xd206470f7c921f7b, 0x3ee4d8465b3ed0b8, 0xb21e6d5f47bdd0a1, 0xa01669445816a58b, 0xc01662c780, 0x0, 0x0)
        github.com/ethereum/go-ethereum/eth/protocols/snap/sync.go:653 +0xc2e
github.com/ethereum/go-ethereum/eth/downloader.(*stateSync).run(0xc02de3d7c0)
        github.com/ethereum/go-ethereum/eth/downloader/statesync.go:318 +0x87
created by github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).runStateSync
        github.com/ethereum/go-ethereum/eth/downloader/statesync.go:114 +0x1ef

```

Tried several times to do fresh sync.
Impossible to sync full node.
Any workaround?

Thanks

**Comment by princesscrypto1337** [comment]
I'm running into the same panic after syncing from scratch with snap mode

```
panic: deletion not supported

goroutine 1531641 [running]:
github.com/ethereum/go-ethereum/trie.(*StackTrie).TryUpdate(0xc053ffcf70, 0xc029140460, 0x20, 0x20, 0x280a478, 0x0, 0x0, 0xc043d8b7e8, 0xc0da2e3e00)
        github.com/ethereum/go-ethereum/trie/stacktrie.go:194 +0xd2
github.com/ethereum/go-ethereum/trie.(*StackTrie).Update(0xc053ffcf70, 0xc029140460, 0x20, 0x20, 0x280a478, 0x0, 0x0)
        github.com/ethereum/go-ethereum/trie/stacktrie.go:201 +0x77
github.com/ethereum/go-ethereum/eth/protocols/snap.(*Syncer).processStorageResponse(0xc00000cf00, 0xc04c478600)
        github.com/ethereum/go-ethereum/eth/protocols/snap/sync.go:1928 +0x3f4
github.com/ethereum/go-ethereum/eth/protocols/snap.(*Syncer).Sync(0xc00000cf00, 0xcf8fef4722d1f77f, 0x1753f77344dc40fd, 0xaf97cc4260622dc9, 0xd2d6d8a0580567aa, 0xc019de17a0, 0x0, 0x0)
        github.com/ethereum/go-ethereum/eth/protocols/snap/sync.go:653 +0xcae
github.com/ethereum/go-ethereum/eth/downloader.(*stateSync).run(0xc0582ffae0)
        github.com/ethereum/go-ethereum/eth/downloader/statesync.go:318 +0x87
created by github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).runStateSync
        github.com/ethereum/go-ethereum/eth/downloader/statesync.go:114 +0x1ef
```

**Comment by smartins** [comment]
Seems to be related to syncing using snap mode, don't get this with the default method but got when using snap

**Comment by Marijus** [comment]
facing same issue - any tips?

**Comment by vrogojin** [comment]
I switched to other physical nvme, and it worked without problem. The first nvme drive itself was fine though. No idea, what geth did not like about it.

**Comment by nedmarinov** [comment]
My node was working perfectly until I upgraded from ssd to nvme. That doesn't make any sense to me but I see that @vrogojin had the same issue with the nvme.
Any update on this?

**Comment by MischaWeerwag** [comment]
Same issue. Looks like it can be fixed by increasing the soft/hard limit of open files from 1024 to 65535.

**Comment by ikgo** [comment]
any status update on the bug.
my node v1.1.5 enter to restart loop. panic after couple of minutes after star syncing.

is a version 1.1.6 fix this bug?

**Comment by seaify** [comment]
when can it be fixed? can't sync

**Comment by KeefeL** [comment]
Fixed by #667, please wait for next release.
